src/CMakeLists.txt

```diff
@@ -65,10 +65,12 @@ add_executable(castxml
   RunClang.cxx RunClang.h
   Utils.cxx Utils.h
   )
-target_link_libraries(castxml
-  ${clang_libs}
-  ${llvm_libs}
-  )
+if(LLVM_LINK_LLVM_DYLIB)
+  set(USE_SHARED USE_SHARED)
+endif()
+llvm_config(castxml ${USE_SHARED} ${llvm_libs})
+target_link_libraries(castxml PRIVATE ${clang_libs})
```

Comment:
When `LLVM_LINK_LLVM_DYLIB` is *not* on I get many `/usr/bin/ld: cannot find -lLLVMLLVM...` errors.

`llvm_config` accepts a list of component names, so they should not go through `llvm_map_components_to_libnames`.  Please revise.

Reply:
Okay, an earlier version of this did not use ```llvm_map_components_to_libnames```. I suppose that's the reason I had that. I'll revise.